payments/views/v0/__init__.py

```diff
@@ -471,3 +473,45 @@ def add_discount_to_basket(request, system_slug: str):
         BasketWithProductSerializer(basket).data,
         status=status.HTTP_200_OK,
     )
+
+
+class DiscountAPIViewSet(APIView):
+    """
+    Provides API for creating Discount objects.
+    Discounts created through this API will be associated
+    with the integrated system that is linked to the api key.
+
+    Responds with a 201 status code if the discount is created successfully.
+    """
+
+    permission_classes = [HasIntegratedSystemAPIKey]
+    authentication_classes = []  # disables authentication
+
+    @extend_schema(
+        description="Create a discount.",
+        methods=["POST"],
+        request=None,
+        responses=DiscountSerializer,
+    )
+    def post(self, request):
+        """
+        Create discounts.
+
+        Args:
+            request: The request object.
+
+        Returns:
+            Response: The response object.
+        """
+        key = request.META["HTTP_AUTHORIZATION"].split()[1]
+        api_key = IntegratedSystemAPIKey.objects.get_from_key(key)
+        discount_dictionary = request.data
+        discount_dictionary["integrated_system"] = str(api_key.integrated_system.id)
+        discount_codes = api.generate_discount_code(
```

Comment:
Ok, one more small thing I forgot about - transaction number should be in here too.